include/dfsinternal.h

```diff
@@ -24,10 +24,12 @@
 /** @brief Representation of a directory entry */
 struct directory_entry
 {
-    /** @brief File size and flags.  See #FLAGS_FILE, #FLAGS_DIR and #FLAGS_EOF */
-    uint32_t flags;
+    /** @brief File size */
+    uint32_t size;
     /** @brief Offset to next directory entry */
     uint32_t next_entry;
+    /** @brief File flags.  See #FLAGS_FILE, #FLAGS_DIR and #FLAGS_EOF */
+    uint8_t flags;
```

Comment:
How does it matter that we changed size and / or alignment of fields here? My assumption is that if you're upgrading libdragon you need to regenerate the DFS anyway (files have changed serialization anyhow), so there's no backward compatibility concern.

Reply:
(BTW, the structure is indeed already packed, this is also why I decreased MAX_FILENAME_LEN by 1 and moved the flags field just before the path).

Reply:
It matters because of wasted space otherwise, as well as possible GCC differences on the host and R4300. AFAIK, struct packing without any directives is undefined in C.